common/content/statusline.js

```diff
@@ -127,17 +127,19 @@ const StatusLine = Module("statusline", {
 
     // set the visibility of the statusline
     setVisibility: function (request) {
-        if ( typeof this.setVisibility.currVisibility == 'undefined' ) {
-            this.setVisibility.currVisibility = true;
-            this.setVisibility.prevVisibility = true;
-            this.setVisibility.contentSeparator
-                = highlight.get('ContentSeparator').value;
-
-            // kinds of requests
-            this.setVisibility.MODE_ON  = 0;      // commandline active
-            this.setVisibility.MODE_OFF = 1;      // commandline inactive
-            this.setVisibility.TOGGLE = 2;        // toggle on or off
-            this.setVisibility.FULLSCREEN = 3;    // in or out of fullscreen
+        if ( typeof this.setVisibility.MODE_AUTO == 'undefined' ) { // TODO proper initialization
```

Comment:
I don't know what's the proper way of initializing public fields of a method in JavaScript (with the class/module/whatever declaration that is used in vimperator), so I kept the old implementation. The problem is, that the fields are not available before the first call of `setVisibility()`.